### PR TITLE
Fix some strings not translatable

### LINKED ISSUE #31849: Some strings not translatable in M2.4.1
### Preconditions (*)
1. Clean Magento 2.4.1 installation
1. 2.4-develop

### Steps to reproduce (*)
1. Add a product to cart
1. Navigate to cart page
1. Open admin interface, find the product, save it as "Out of Stock"
1. Refresh cart page, see error message: **There are no source items with the in stock status**
1. Find the translation for the message in <magento install dir>vendor/magento/module-catalog-inventory/i18n/en_US.csv
1. Change the result string to be something else
1. Clean "translate" cache
1. Refresh the cart page and witness the previously mentioned string being not translated to value we changed it at point 6

### Expected result (*)
1. Error messages are translated

### Actual result (*)
1. Error messages are not translated - can be verified even with default en_US language - change the translation for the error message

### NOTES
The **steps to reproduce** part only highlights one specific string, however this seems to be a more general problem:

Due to update of design (and translations) loading in https://github.com/magento/magento2/commit/b2a035d93c98487b69bbcd6b719e79ddf8e39574
(loading design and translations was moved from beforeDispatch to beforeExecute)

However there are several plugins in M2 ecosystem, which come before actual execute call, \Magento\InventorySales\Model\IsProductSalableForRequestedQtyCondition\IsProductSalableForRequestedQtyConditionChain is one of the many examples (on quote loading, validation is triggered).

Verified this in clean Magento 2.4.1 - the error phrase is never translated, as renderers for Phrase is set only in \Magento\Theme\Plugin\LoadDesignPlugin::beforeExecute

This happens due to renderer not being set to \Magento\Framework\Phrase, and if it is used earlier than the renderers are set - the original string gets returned, as it uses the RendererPlaceholder class to render them.

With current state, a lot of strings cannot be translated, the only imaginable workaround is to move design and translations loading back to beforeDispatch method. (Moving just the translation loading to beforeDispatch won't help, if the translations are in theme.)

## 
Please provide [Severity](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- < ] Severity: **S0** *- Affects critical data or functionality and leaves users without workaround.*
- [ ] Severity: **S1** *- Affects critical data or functionality and forces users to employ a workaround.*
- [x] Severity: **S2** *- Affects non-critical data or functionality and forces users to employ a workaround.*
- [ ] Severity: **S3** *- Affects non-critical data or functionality and does not force users to employ a workaround.*
- [ > Severity: **S4** *- Affects aesthetics, professional look and feel, “quality” or “usability”.*

### PR DESCRIPTION
**Pull request by mrtuvn**
### Description (*)

Init part theme and translate in very first dispatch from FrontController
https://devdocs.magento.com/guides/v2.4/extension-dev-guide/routing.html

One main archieve in this PR is to reduce frequence high load from design plugin `designLoader` in 

https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/Theme/etc/di.xml#L109 by move load parts to FrontController instead called in plugin beforeExecute. Only keep on line for check design change

Another approach possible to do is use Observer we listens event `controller_action_predispatch`
load design + translate when event fired

### Fixed Issues (if relevant)

1. Fixes magento/magento2#31849

### Manual testing scenarios (*)

1. Following steps in report issues

### Questions or comments

CC: @siimm @lbajsarowicz @Vinai @sidolov @gabrieldagama @zakdma 

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/Theme/etc/di.xml [hunk 1]
```diff
@@ -105,9 +105,6 @@
             <argument name="scope" xsi:type="const">Magento\Store\Model\ScopeInterface::SCOPE_STORE</argument>
         </arguments>
     </virtualType>
-    <type name="Magento\Framework\App\ActionInterface">
-        <plugin name="designLoader" type="Magento\Theme\Plugin\LoadDesignPlugin"/>
```
Comment (siimm) [inline app/code/Magento/Theme/etc/di.xml]: The class `Magento\Theme\Plugin\LoadDesignPlugin` can probably removed alltogether after this change ?
As I see this was the only use of that class.

Comment (mrtuvn) [reply app/code/Magento/Theme/etc/di.xml]: Sure yes you can do that

Comment (mrtuvn) [reply app/code/Magento/Theme/etc/di.xml]: In fact we should disable or deprecate this file instead completely remove it for maintain backward compatible

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run Functional Tests B2B

**Comment by mrtuvn** [comment]
Seem this PR fix problem reported by you @siimm
https://github.com/magento/magento2/pull/31970/

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
Closing this since problem fixed by another pull request mentioned previous above

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run all tests

File: lib/internal/Magento/Framework/View/DesignLoader.php [hunk 2]
```diff
@@ -51,8 +51,10 @@ public function __construct(
...
         $area = $this->_areaList->getArea($this->appState->getAreaCode());
-        $area->load(\Magento\Framework\App\Area::PART_DESIGN);
-        $area->load(\Magento\Framework\App\Area::PART_TRANSLATE);
-        $area->detectDesign($this->_request);
+        if ($area &&
+            $this->_request &&
+            $this->_request instanceOf RequestInterface) {
+            $area->detectDesign($this->_request);
+        }
     }
```
Comment (mrtuvn) [inline lib/internal/Magento/Framework/View/DesignLoader.php:59]: @Vinai @lbajsarowicz 
moved load part to FrontController to reduce frequently load to much

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run all tests

File: lib/internal/Magento/Framework/App/FrontController.php [hunk 3]
```diff
@@ -137,6 +137,9 @@ public function dispatch(RequestInterface $request)
         $this->validatedRequest = false;
         $routingCycleCounter = 0;
         $result = null;
+        $area = $this->areaList->getArea($this->appState->getAreaCode());
+        $area->load(Area::PART_DESIGN);
+        $area->load(Area::PART_TRANSLATE);
         while (!$request->isDispatched() && $routingCycleCounter++ < 100) {
```
Comment (mrtuvn) [inline lib/internal/Magento/Framework/App/FrontController.php]: The only concern here is at the moment run into this method `dispatch` some reason what if `$area` not available ?

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run Functional Tests EE , Functional Tests CE, Functional Tests B2B

**Comment by mrtuvn** [comment]
@engcom-Alfa can you check with this change ?

**Comment by mrtuvn** [comment]
@magento run Functional Tests EE, Functional Tests B2B

**Comment by mrtuvn** [comment]
@magento run Functional Tests EE, Functional Tests B2B

**Comment by mrtuvn** [comment]
@magento run Functional Tests EE, Functional Tests B2B

**Comment by mrtuvn** [comment]
@magento run Functional Tests EE, Functional Tests B2B

**Comment by mrtuvn** [comment]
closed this PR when origin issue can't reproduce

**Pull request closed without merging**